src/GlucoPilot.Api/Endpoints/Readings/List.cs

```diff
@@ -0,0 +1,48 @@
+﻿using FluentValidation;
+using GlucoPilot.Api.Models;
+using GlucoPilot.Data;
+using GlucoPilot.Data.Entities;
+using GlucoPilot.Data.Repository;
+using GlucoPilot.Identity.Authentication;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.HttpResults;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace GlucoPilot.Api.Endpoints.Readings;
+
+internal static class List
+{
+    internal static async Task<Results<Ok<List<ReadingsResponse>>, ValidationProblem>> HandleAsync(
+        [FromQuery] ListReadingsRequest request,
```

Comment:
I think this is what you want:
```suggestion
        [AsParameters] ListReadingsRequest request,
```